lib/Conversion/TorchToLinalg/Linear.cpp

```diff
@@ -549,32 +549,51 @@ class ConvertAtenConvolutionOp : public OpConversionPattern<AtenConvolutionOp> {
                                          "only support constant int dilations");
 
     Value N = getDimOp(rewriter, loc, input, 0);
+    Value IC = getDimOp(rewriter, loc, input, 1);
     SmallVector<Value> inDims;
     for (size_t i = 2; i < inRank; i++)
       inDims.push_back(getDimOp(rewriter, loc, input, i));
     Value F = getDimOp(rewriter, loc, weight, 0);
+    Value WC = getDimOp(rewriter, loc, weight, 1);
     SmallVector<Value> weightDims;
     for (size_t i = 2; i < inRank; i++)
       weightDims.push_back(getDimOp(rewriter, loc, weight, i));
 
-    // Guard unused values (transposed, groups)
-    int64_t group_size;
-    if (!matchPattern(op.groups(), m_TorchConstantInt(&group_size)) ||
-        group_size != 1)
-      return rewriter.notifyMatchFailure(
-          op, "unimplemented: only group size of 1 supported");
+    // Guard unused values (transposed)
     bool transposed = true;
     if (!matchPattern(op.transposed(), m_TorchConstantBool(&transposed)) ||
         transposed)
       return rewriter.notifyMatchFailure(
           op, "unimplemented: only non-transposed convolution supported");
 
+    // Checks for valid group size
+    int64_t group_size;
+    if (!matchPattern(op.groups(), m_TorchConstantInt(&group_size)))
+      return rewriter.notifyMatchFailure(op,
+                                         "only constant group size supported");
+    Value groups = castIntToIndex(rewriter, loc, adaptor.groups());
+    Value groupInputValid = rewriter.create<arith::CmpIOp>(
+        loc, arith::CmpIPredicate::eq, IC,
+        rewriter.create<arith::MulIOp>(
+            loc, groups,
+            rewriter.create<arith::FloorDivSIOp>(loc, IC, groups)));
+    rewriter.create<cf::AssertOp>(
+        loc, groupInputValid,
+        rewriter.getStringAttr(
+            "invalid: groups must divide input channel size evenly"));
+    Value weightInputValid = rewriter.create<arith::CmpIOp>(
+        loc, arith::CmpIPredicate::eq, F,
+        rewriter.create<arith::MulIOp>(
+            loc, groups, rewriter.create<arith::FloorDivSIOp>(loc, F, groups)));
+    rewriter.create<cf::AssertOp>(
+        loc, weightInputValid,
+        rewriter.getStringAttr(
+            "invalid: groups must divide weight batch size evenly"));
+
     // Pad the input tensor according to padding.
     SmallVector<int64_t, 4> paddingIncludingNC = {0, 0};
     paddingIncludingNC.insert(paddingIncludingNC.end(), paddingInts.begin(),
                               paddingInts.end());
```

Comment:
Can you also move this down to where you moved `paddedInput` since it is only used there? Also, I think you can do `padingIncludingNC.append(paddingInts)`

Reply:
Do you mean where it is now? I don't want to put it in the loop.

Reply:
Yeah, that works!

Reply:
Yeah, that works!